.ci/Jenkinsfile

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env groovy
+
+@Library('apm@current') _
+
+pipeline {
+  agent { label 'ubuntu-18 && immutable' }
+  environment {
+    BASE_DIR="src/github.com/elastic/package-spec"
+    JOB_GIT_CREDENTIALS = "f6c7695a-671e-4f4f-a331-acdce44ff9ba"
+    PIPELINE_LOG_LEVEL='INFO'
+  }
+  options {
+    timeout(time: 15, unit: 'MINUTES')
+    buildDiscarder(logRotator(numToKeepStr: '20', artifactNumToKeepStr: '20', daysToKeepStr: '30'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+    rateLimitBuilds(throttle: [count: 60, durationName: 'hour', userBoost: true])
+    quietPeriod(10)
+  }
+  triggers {
+    issueCommentTrigger('(?i)(.*(?:jenkins\\W+)?run\\W+(?:the\\W+)?tests(?:\\W+please)?.*|^\\/test$)')
+  }
+  stages {
+    /**
+     Checkout the code and stash it, to use it on other stages.
+     */
+    stage('Checkout') {
+      steps {
+        deleteDir()
+        gitCheckout(basedir: "${BASE_DIR}")
+        stash allowEmpty: true, name: 'source', useDefaultExcludes: false
+      }
+    }
+    /**
+     Check the source code.
+     */
+    stage('Check') {
+      steps {
+        cleanup()
+        withMageEnv(){
+          dir("${BASE_DIR}"){
+            sh(label: 'Check',script: 'make check')
+          }
+        }
+      }
+    }
+    /**
+     Test the source code.
+     */
+    stage('Test') {
+      steps {
+        cleanup()
+        withMageEnv(){
+          dir("${BASE_DIR}"){
+            sh(label: 'Check',script: 'make test')
```

Comment:
What output does `make test` produce? Is it something that needs to be processed with `junit`?

Reply:
At the moment `make test` will simply run `go test -v ./...` under the hood: https://github.com/elastic/package-spec/pull/12/files#diff-d635a9f292641eeeb317ef8c83354e7dR16

Would it be better if it output in a junit-compatible format? We could certainly make that happen with something like https://github.com/jstemmer/go-junit-report.